lib/google/cloud/gemserver/cli/server.rb

```diff
@@ -75,56 +78,76 @@ def deploy
           end
 
           ##
-          # Deletes a given gemserver by its parent project's ID.
+          # Deletes a given gemserver and its Cloud SQL instance
           #
           # @param [String] proj_id The project ID of the project the gemserver
           # was deployed to.
           def delete proj_id
-            return unless Configuration.deployed?
-            full_delete = user_input("This will delete the entire Google Cloud"\
-               " Platform project #{proj_id}. Continue"\
-               " deletion? (Y|n, default n) If no, all relevant resources will"\
-               " be deleted besides the parent GCP project.").downcase
-            if full_delete == "y"
-              puts "Deleting gemserver with parent project"
-              system "gcloud projects delete #{proj_id}"
+            return unless @config.deployed?
+            if @config.metadata[:platform] == "gae"
+              full_delete = user_input("This will delete the entire Google Cloud"\
+                " Platform project #{proj_id}. Continue"\
+                " deletion? (Y|n, default n) If no, all relevant resources will"\
+                " be deleted besides the parent GCP project.").downcase
+              if full_delete == "y"
+                puts "Deleting gemserver with parent project"
+                system "gcloud projects delete #{proj_id}"
+              else
+                @config.delete_from_cloud
+                del_gcs_files
+                puts "Visit:\n https://console.cloud.google.com/appengine/"\
+                  "settings?project=#{proj_id} and click \"Disable "\
+                  " Application\" to delete the Google App Engine application"\
+                  " the gemserver was deployed to."
+              end
             else
+              name = user_input "Enter the name of the container cluster"
+              zone = user_input "Enter the zone of the cluster"
+              system "kubectl delete service #{Deployer::IMAGE_NAME}"
+              system "kubectl delete deployment #{Deployer::IMAGE_NAME}"
+              system "gcloud container clusters delete #{name} -z #{zone}"
               @config.delete_from_cloud
               del_gcs_files
-              inst = @config.app["beta_settings"]["cloud_sql_instances"]
-                .split(":").pop
-              puts "Deleting child Cloud SQL instance #{inst}..."
-              params = "delete #{inst} --project #{proj_id}"
-              status = system "gcloud beta sql instances #{params}"
-              fail "Unable to delete instance" unless status
-              puts "The Cloud SQL instance has been deleted. Visit:\n "\
-                "https://console.cloud.google.com/appengine/settings?project="\
-                "#{proj_id} and click \"Disable Application\" to delete the "\
-                "Google App Engine application the gemserver was deployed to."
             end
+
+            inst = @config.app["beta_settings"]["cloud_sql_instances"]
```

Comment:
This is GAE specific. Changing this to `@config.config[:db_connection_options][:socket].split(":").pop` is a better solution (extracts the Cloud SQL instance name from the required `config.yml`).